1_fetch/src/get_nwis_data.R

```diff
@@ -1,160 +1,151 @@
-                                Date=col_date(format="%Y-%m-%d"),
-                                discharge=col_double(),
-                                discharge_cd=col_character()))
-  data<-addWaterYear(data)
+  filepath <- filenames[grep(site, filenames)]
+  data <- read_csv(filepath,
+                   col_types=cols(agency_cd=col_character(),
+                                  site_no=col_character(),
+                                  Date=col_date(format="%Y-%m-%d"),
+                                  discharge=col_double(),
+                                  discharge_cd=col_character()))
+  data <- addWaterYear(data)
   ##remove all data from years with data gaps
-  keep_years<-screen_daily_flow %>%
-    filter(site_no ==site)%>%
+  keep_years <- screen_daily_flow %>%
+    filter(site_no == site) %>%
     pull(complete_yrs)
-  if (yearType=="water"){
-    data_sc<-data[which(data$waterYear %in% keep_years),]
-  } else{
-    data_sc<-data[which(year(data$Date)%in% keep_years),]
+  if (yearType == "water"){
+    data_sc <- data[which(data$waterYear %in% keep_years), ]
+  }else{
+    data_sc <- data[which(year(data$Date) %in% keep_years), ]
   }
-
-  df<-data.frame(date=as.Date(data_sc$Date),discharge=data_sc$discharge)
+  
+  df <- data.frame(date=as.Date(data_sc$Date), discharge=data_sc$discharge)
   ###run EflowStats validation to produce clean, ready to process data
-  clean_data<-validate_data(df,yearType=yearType)
-  clean_data$site_no<-unique(data_sc$site_no)
+  clean_data <- validate_data(df, yearType=yearType)
+  clean_data$site_no <- unique(data_sc$site_no)
   return(clean_data)
-}  #end clean_daily_data function
+}#end clean_daily_data function
 
-
-get_NWIS_drainArea<-function(site_num){
+get_NWIS_drainArea <- function(site_num){
   print(site_num)
-  data_out<- data.frame(site_no=as.character(site_num),
-                        drainArea=readNWISsite(siteNumbers=site_num)$drain_area_va)
+  data_out <- data.frame(site_no=as.character(site_num),
+                         drainArea=readNWISsite(siteNumbers=site_num)$drain_area_va)
   return(data_out)
 }
 
-
-get_floodThreshold<-function(site_num,p1_clean_daily_flow,p1_peak_flow,perc,yearType){
+get_floodThreshold <- function(site_num, p1_clean_daily_flow, p1_peak_flow, 
+                               perc, yearType){
   print(site_num)
-  df_dv<- p1_clean_daily_flow %>%
-    filter(site_no==site_num)%>%
-    select(date,discharge)
+  df_dv <- p1_clean_daily_flow %>%
+    filter(site_no == site_num) %>%
+    select(date, discharge)
   
-  filepath<-p1_peak_flow[grep(site_num,p1_peak_flow)]
-  peaks<-read_csv(filepath) 
-  peaks$site_no<-as.character(peaks$site_no)
-  df_pk<-data.frame(date=as.Date(peaks$peak_dt),peak=peaks$peak_va)
-  floodThreshold<-get_peakThreshold(df_dv,df_pk,perc=perc,yearType=yearType)
-  df_out<-data.frame(site_no=site_num,floodThreshold)
+  filepath <- p1_peak_flow[grep(site_num, p1_peak_flow)]
+  peaks <- read_csv(filepath, show_col_types = FALSE) 
```

Comment:
`show_col_types = FALSE` caused an error on my end. Maybe a version issue? Not sure. 

I replaced with `col_types = cols()` and it ran successfully.

Reply:
Interesting - that error does not appear for me. I'll change to `col_types = cols()` to be safe. That is the default option and it also suppressed the printed info for me.